### ISSUE TITLE
0.2.0 Discussion

### ISSUE THREAD

**Issue by leeola** [leeola/appaid.kdapp#1]
Questions, comments, or desires for the testing-oriented 0.2.0 release should be posted here.

**Comment by stefanbc** [comment]
How about a clear/close button for the inception cases?

**Comment by leeola** [comment]
Good idea! though i would classify that as a 0.1.0 feature enhancement, i'll open a ticket for it, thanks :)

Until i add that, to work around this you can simply re-load the root app. :)

**Comment by leeola** [comment]
@cliffrowley Do you have any preferences on a Browser testing framework? I'd prefer something not too opinionated (the framework itself), letting people work how they like.

**Comment by cliffrowley** [comment]
Personally, I use Capybara, but it's Ruby so I'm not sure how useful it'd be here unless you write your tests in Ruby too (e.g. minitest or rspec).  Capybara will allow you to abstract the browser, supporting Selenium, ChromeDriver, QTWebKit, or Poltergeist - or actually any browser you want with a little (not much) work.  It's pretty good.

I don't mind helping put something together.

**Comment by leeola** [comment]
Well for starters, i think we need to answer the following:

Do we want api/framework only tests? Or do we want browser emulation-like tests (Selenium/etc). I lean towards the first, but if we can support both nicely, without making this heavy, clunky, unfriendly, and basically bad, then i'd like to support both.

I tend to dislike a lot of browser automatic tools, and prefer AngularJS-like approaches with more mock focused/supportive APIs, but that's not denying the usefulness of real browser automation.

I do think ideally, we will use a JavaScript-language framework, as CoffeeScript(Or JS) is what the majority of KDApps will be written in

**Comment by cliffrowley** [comment]
Do we want api/framework only tests? Or do we want browser emulation-like tests (Selenium/etc). I lean towards the first, but if we can support both nicely, without making this heavy, clunky, unfriendly, and basically bad, then i'd like to support both.

I tend to dislike a lot of browser automatic tools, and prefer AngularJS-like approaches with more mock focused/supportive APIs, but that's not denying the usefulness of real browser automation.

No idea, you asked about browser based testing, I told you what I use ;-)

**Comment by leeola** [comment]
> No idea, you asked about browser based testing, I told you what I use ;-)

Well when i said browser based, i meant.. can run in browser. That's the only location _for_ the API, so even if we have framework only tests, it's still browser based. I think there are two different types, which i pointed out. Simple JS ones that are generally designed for Node, but also run in the browser, don't emulate real browser junk. Selenium/etc are in another category haha.

Anyway, i figured you would have an opinion on the choice between the two :)

**Comment by leeola** [comment]
Also, email replies are horrible at quoting. lol.

**Comment by cliffrowley** [comment]
I guess don't really an opinion on a client side testing framework because I don't really do it - I'm a server side guy with an overlap in front end tech when it comes to web stuff :-)
